### PR TITLE
NpmAudit task

### PR DESCRIPTION
**Pull request by splatteredbits**

### CODE REVIEW

File: Whiskey/Tasks/Invoke-WhiskeyDotNetBuild.ps1 [hunk 1]
```diff
@@ -17,7 +17,9 @@ function Invoke-WhiskeyDotNetBuild
     Set-StrictMode -Version 'Latest'
     Use-CallerPreference -Cmdlet $PSCmdlet -SessionState $ExecutionContext.SessionState
 
-    $dotnetExe = $TaskParameter['DotNetPath']
+    Write-Warning -Message ('The "DotNetTest" task is obsolete and will be removed in a future version of Whiskey. Please use the "DotNet" task instead.')
+
+    $dotnetExe = $TaskParameter['DotNetBuild']
```
Comment (DecoyJoe) [inline Whiskey/Tasks/Invoke-WhiskeyDotNetBuild.ps1]: This should still be `$TaskParameter['DotNetPath']`

File: Whiskey/Tasks/Invoke-WhiskeyDotNetPack.ps1 [hunk 2]
```diff
@@ -17,7 +17,9 @@ function Invoke-WhiskeyDotNetPack
     Set-StrictMode -Version 'Latest'
     Use-CallerPreference -Cmdlet $PSCmdlet -SessionState $ExecutionContext.SessionState
 
-    $dotnetExe = $TaskParameter['DotNetPath']
+    Write-Warning -Message ('The "DotNetTest" task is obsolete and will be removed in a future version of Whiskey. Please use the "DotNet" task instead.')
+
+    $dotnetExe = $TaskParameter['DotNetPack']
```
Comment (DecoyJoe) [inline Whiskey/Tasks/Invoke-WhiskeyDotNetPack.ps1]: This should still be `$TaskParameter['DotNetPath']`

File: Whiskey/Tasks/Invoke-WhiskeyNodeTask.ps1 [hunk 3]
```diff
@@ -24,7 +24,7 @@ function Invoke-WhiskeyNodeTask
     Set-StrictMode -Version 'Latest'
     Use-CallerPreference -Cmdlet $PSCmdlet -SessionState $ExecutionContext.SessionState
 
-    Write-Warning -Message ('The ''Node'' task has been deprecated and will be removed in a future version of Whiskey. It''s functionality has been broken up into individual smaller tasks, ''NpmInstall'', ''NpmRunScript'', ''NodeNspCheck'', and ''NodeLicenseChecker''. Update the build configuration in ''{0}'' to use the new tasks.' -f $TaskContext.ConfigurationPath)
+    Write-Warning -Message ('The "Node" task is obsolete and will be removed in a future version of Whiskey. It''s functionality has been broken up into individual smaller tasks, "NpmInstall", "NpmRunScript", "NpmAudit", and "NodeLicenseChecker". Update the build configuration in "{0}" to use the new tasks.' -f $TaskContext.ConfigurationPath)
```
Comment (DecoyJoe) [inline Whiskey/Tasks/Invoke-WhiskeyNodeTask.ps1]: This warning should be updated considering the recommended tasks that it references are now also deprecated.

File: Whiskey/Whiskey.psd1 [hunk 4]
```diff
@@ -150,6 +150,11 @@
             # ReleaseNotes of this module
             ReleaseNotes = @'
 * `Version` task now supports pulling a version number from a Chef Cookbook `metadata.rb` file.
+* Created `Npm` task for running NPM commands.
+* Deprecated the `NpmConfig`, `NpmInstall`, `NpmPrune`, and `NpmRunScript` tasks. They will be removed in a future version of Whiskey. Use the new `Npm` task instead.
+* The `NodeNspCheck` task is obsolete. Migrate to the `NpmAudit` task.
```
Comment (DecoyJoe) [inline Whiskey/Whiskey.psd1]: This line is incorrect, the `NpmAudit` task does not exist.

File: Whiskey/en-US/about_Whiskey_DotNet_Task.help.txt [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+    Runs dotnet.exe commands.
+
+DESCRIPTION
+    The `DotNet` tasks runs dotnet.exe commands. .NET Core is installed for you into a ".dotnet" directory (in the directory of your whiskey.yml file). You can control what version of the .NET Core SDK to download and use with the "SdkVersion" parameter.
+    
+    If a command accepts any paths to files, pass them to the `Path` property. Wildcards are allowed and the `DotNet` task will run the dotnet.exe command once for each path.
+
+    Pass any command arguments with the `Argument` property.
+
+    The task writes a detailed build log to the build output directory. If the `Path` property has no value, the file is named `dotnet.COMMAND.log`. Otherwise, there will be a log file for each path named `dotnet.build.FILE_NAME.log`, where `FILE_NAME` is the name poertion of the path.
```
Comment (DecoyJoe) [inline Whiskey/en-US/about_Whiskey_DotNet_Task.help.txt]: `dotnet.build.FILE_NAME.log` should be `dotnet.COMMAND.FILE_NAME.log`

Also typo on `poertion`

File: Whiskey/en-US/about_Whiskey_Npm_Task.help.txt [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+DESCRIPTION
+    The `Npm` task runs NPM commands. Pass the name of the command to the `Command` property. Pass command arguments to the `Argument` property. If the NPM command returns a non-zero exit code, the build will fail.
+
+    By default, uses the latest LTS version of Node and the version of NPM that ships with it. To customize the version of Node to use, specify that version in the `engines.node` property in the package.json file or use the task's `NodeVersion` property. The task looks for a package.json file in the its working directory, which by default is the directory of your whiskey.yml file. You can change the task's working directory with the `WorkingDirectory` property.
+
+    You can customize the version of NPM to use with the `NpmVersion` property.
+
+PROPERTIES
+
+    * `WorkingDirectory`: the directory where the `package.json` exists. Defaults to the directory where the build's `whiskey.yml` file was found. Must be relative to the `whiskey.yml` file.
```
Comment (DecoyJoe) [inline Whiskey/en-US/about_Whiskey_Npm_Task.help.txt]: Nit: This is a common task property that is documented in `about_Whiskey_Tasks`

File: Whiskey/en-US/about_Whiskey_Npm_Task.help.txt [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+
+EXAMPLES
+
+    ## Example 1
+
+        Build:
+        - Npm:
+            Command: install
+
+    Demonstrates how to runs `npm install` to download and install your project's dependencies.
```
Comment (DecoyJoe) [inline Whiskey/en-US/about_Whiskey_Npm_Task.help.txt]: Typo: "how to **runs**" should be "how to **run**"

**Comment by splatteredbits** [comment]
@DecoyJoe Ready for re-review.

File: Whiskey/Whiskey.psd1 [hunk 8]
```diff
@@ -150,6 +150,11 @@
 * Added support for PowerShell modules to the RequiresTool attribute.
 * Whiskey tasks that use PowerShell modules now use Whiskey's auto-download feature to download required modules during the build.
 * Fixed: publishing to the PowerShell Gallery released in September 2018 is failing. Updating to PowerShellGet 2.0.0 and PackageManagement 1.1.7.2.
+* Created `Npm` task for running NPM commands.
+* Deprecated the `NpmConfig`, `NpmInstall`, `NpmPrune`, and `NpmRunScript` tasks. They will be removed in a future version of Whiskey. Use the new `Npm` task instead.
+* The `NodeNspCheck` task is obsolete. Migrate to the `NpmAudit` task.
```
Comment (DecoyJoe) [inline Whiskey/Whiskey.psd1]: `NpmAudit` isn't a real task, is it?

File: Whiskey/Tasks/Invoke-WhiskeyDotNetBuild.ps1 (hunk 1 above)
Comment (DecoyJoe) [reply Whiskey/Tasks/Invoke-WhiskeyDotNetBuild.ps1]: This still needs to be fixed.

File: Whiskey/Tasks/Invoke-WhiskeyDotNetPack.ps1 (hunk 2 above)
Comment (DecoyJoe) [reply Whiskey/Tasks/Invoke-WhiskeyDotNetPack.ps1]: This still needs to be fixed.

File: Whiskey/en-US/about_Whiskey_Npm_Task.help.txt (hunk 7 above)
Comment (DecoyJoe) [reply Whiskey/en-US/about_Whiskey_Npm_Task.help.txt]: Still a typo here

File: Whiskey/en-US/about_Whiskey_DotNet_Task.help.txt (hunk 5 above)
Comment (DecoyJoe) [reply Whiskey/en-US/about_Whiskey_DotNet_Task.help.txt]: Still needs fixing.

**Comment by splatteredbits** [comment]
@DecoyJoe Oops. Totally missed some of your review. Addressed and ready for re-review.

**Comment by splatteredbits** [comment]
@DecoyJoe Fixed `Invoke-WhiskeyDotNetPack.ps1`. Build is failing for what looks like an unrelated reason. I'm trying to figure out what's going on.

**Comment by splatteredbits** [comment]
@DecoyJoe Build fixed. I'm ready to merge.

**Pull request merged**